pkg/profiler/cpu/maps.go

```diff
@@ -138,3 +144,92 @@ func (m *bpfMaps) clean() error {
 
 	return nil
 }
+
+// setUnwindTable updates the unwind tables with the given unwind table.
+func (m *bpfMaps) setUnwindTable(pid int, ut unwind.UnwindTable) error {
+	buf := new(bytes.Buffer)
+
+	// Write number of rows `.table_len``.
+	if err := binary.Write(buf, m.byteOrder, uint64(len(ut))); err != nil {
+		return fmt.Errorf("write the number of rows: %w", err)
+	}
+
+	if len(ut) >= maxUnwindTableSize {
+		panic(fmt.Sprintf("max unwind table size reached. Table size %d, but max size is %d", len(ut), maxUnwindTableSize))
+	}
+
+	for _, row := range ut {
+		// Right now we only support x86_64, where the return address position
+		// is specified in the ABI, so we don't write it.
+
+		// Write Program Counter (PC).
+		if err := binary.Write(buf, m.byteOrder, row.Loc); err != nil {
+			return fmt.Errorf("write the program counter: %w", err)
+		}
+
+		// Write CFA.
+		switch row.CFA.Rule {
+		case frame.RuleCFA:
+			// Write CFA register.
+			if err := binary.Write(buf, m.byteOrder, row.CFA.Reg); err != nil {
+				return fmt.Errorf("write CFA register bytes: %w", err)
+			}
+
+			// Write CFA offset.
+			if err := binary.Write(buf, m.byteOrder, row.CFA.Offset); err != nil {
+				return fmt.Errorf("write CFA offset bytes: %w", err)
+			}
+		case frame.RuleExpression:
+			// Hack(javierhonduco). Expressions aren't really implemented yet, so let's set some sentinel
+			// values that we can use in the unwinder to detect when we should be using an expression.
+
+			// Write "fake" register.
+			if err := binary.Write(buf, m.byteOrder, uint64(0xBEEF)); err != nil {
+				return fmt.Errorf("write CFA Reg bytes: %w", err)
+			}
+
+			// Write "fake" offset.
+			if err := binary.Write(buf, m.byteOrder, uint64(0xBADFAD)); err != nil {
+				return fmt.Errorf("write CFA offset bytes: %w", err)
+			}
+		default:
+			panic("CFA rule is not valid")
+		}
+
+		// Write $rbp offset.
+		if err := binary.Write(buf, m.byteOrder, row.RBP.Offset); err != nil {
+			return fmt.Errorf("write RBP offset bytes: %w", err)
+		}
+	}
+
+	// Set PID -> unwind table.
+	if err := m.unwindTables.Update(unsafe.Pointer(&pid), unsafe.Pointer(&buf.Bytes()[0])); err != nil {
+		return fmt.Errorf("update unwind tables: %w", err)
+	}
+
+	// HACK(javierhonduco): remove this.
```

Comment:
Leaving this as it's quite useful during debugging, later on, I will add proper logging